storage/src/system/auction/detail.rs

```diff
@@ -567,6 +590,28 @@ where
     }
 }
 
+/// Returns current `ValidatorBid` in case the public key was changed.
+pub fn read_current_validator_bid<P>(
+    provider: &mut P,
+    bid_key: &Key,
+) -> Result<Box<ValidatorBid>, Error>
+where
+    P: StorageProvider + ?Sized,
+{
+    if !bid_key.is_bid_addr_key() {
+        return Err(Error::InvalidKeyVariant);
+    }
+    match provider.read_bid(bid_key)? {
```

Comment:
What do you think would be a reasonable iteration limit?
I expect we can safely assume that actual bids won't form very long chains, so there's no need for additional validation when creating the bridge record. Is that ok?

Reply:
> What do you think would be a reasonable iteration limit?

Considering that's just so we don't do possibly unbounded computation, I think 10 should be more than enough. If there's a legitimate use case that you have to change the keys multiple times within an era then it should serve the purpose

Reply:
I think the longest possible chain could happen when processing unbonds (due to the length of time between creating and processing the `UnbondingPurse`), but I guess it's hard to see a valid use-case for transfering a bid record that often, so 10 iterations should suffice. We'll just have to be clear about this limitation in the documentation. 
On the other hand I'm a bit worried about allowing the user to create an unprocessable entity. Is there a precedent for this in the codebase?

While figuring out if it's possible to create a loop in bridge records I also realized there's another issue with our approach - since bridge records are meant to reuse the same address as `ValidatorBid` it's currently not possible to transfer a bid back to the original address unless the bridge record is pruned. Is this something we should consider as a bug or leave it as a known limitation?

Reply:
Hi @mpapierski.
I've introduced the changes we've discussed:
- bumped the cost of the new entrypoint (to twice the cost of `add_bid` for now)
- removed recursion in favor of a `for` loop with a limited number of iterations